### PR TITLE
Dissolve for idle disp and more

### PR DESCRIPTION
**Pull request by Booplicate**
I wanted to add dissolves for the idle disp from the get go, but didn't have time for it, so I'm doing it now. With these changes (and ideally with an r7 upgrade) all Monika's sprites will use a nice, smooth transition.

### Changes:
- `Dissolve` transition for Monika's idle displayable. The timings are: `0.25` sec for a pose change, `0.1` sec for an exp change.
- New displayables to handle blinking, winks, and tears. That's because the old ATL were clunky and had numerous bugs. The blinking timings are randomized now.
- `Dissolve` transitions for blinking, winks, and tears. The duration for all of them is `0.1` sec (for many of those we can't use more than that, although we could use different durations if we wanted to, leaving this to the testers).
- Some methods of `MASMoniIdleDisp` got the new `skip_dissolve` parameter. It allows us to skip the transition during the next exp change.
- Test cases for the new displayables.
- A displayable to handle following sprites. Supports right/left variants, has commented code for up/down variants too. Also utilises `Dissolve`s and is more optimized than the current `ConditionSwitch`. NOTE: uses a method from #7006.

### Fixes:
- Fixed a bug with incorrect render size of `MASMonikaRender`.
- Fixed a bug with the broken wink ATL (because I deprecated it lol).
- ~~We don't use `scene` in the `spaceroom` label anymore. RenPy has a bug with images being used in both `old_widget` and `new_widget` of a transform (`Dissolve` in this case). Since `scene` is hiding images, it adds them to `old_widget`, then we're showing them, it adds them to `new_widget`. Because of that the images are getting `st` and `at` for both states at once.~~

### Testing:
- Verify the spaceroom looks good (at startup, during idle, during different transitions).
- Check out the new test cases, you can use them for testing.
- Verify the transitions look good. Not too fast, not too slow, no visual bugs.
- Verify blinking, winks, and streaming tears look good both when used in the idle disp and when `show`en directly.
- Verify the transition from idle to a topic and from a topic back to idle (ideally w/ and w/o forcing an exp at the end) looks good (no seizures or weird movements).
- Verify the top of Monika's head doesn't disappear anymore.
- Verify the following sprites look good. Check the transition from a following to non-following and vice versa.

### CODE REVIEW

File: Monika After Story/game/script-ch30.rpy [hunk 1]
```diff
@@ -816,10 +813,12 @@ label spaceroom(start_bg=None, hide_mask=None, hide_monika=False, dissolve_all=F
 
         day_mode = mas_current_background.isFltDay()
 
-    if scene_change:
-        scene black
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-ch30.rpy:820]: the `scene` statement is an easy way to remove all displayables quickly. Much better than having explicit hide statements like the ones you ended up adding elsewhere. If a scene change occurs, we only need to consider what we want to show, not what needs to be hidden.

**Review by ThePotatoGuy — CHANGES_REQUESTED** [review]

File: Monika After Story/game/script-ch30.rpy (hunk 1 above)
Comment (Booplicate) [reply Monika After Story/game/script-ch30.rpy]: Works perfectly now with #7036.

**Review by ThePotatoGuy — CHANGES_REQUESTED** [review]
got this traceback while running but pretty sure its related to something else. 
```
While running game code:
  File "game/script-ch30.rpy", line 1513, in script
    $ mas_randchat.wait()
  File "game/script-ch30.rpy", line 1513, in <module>
    $ mas_randchat.wait()
  File "game/definitions.rpy", line 8273, in wait
    renpy.pause(WAITING_TIME, hard=True)
  File "game/sprite-chart.rpy", line 8205, in render
    img_render = renpy.render(self.current_img, width, height, self.current_st, at)
  File "game/sprite-chart.rpy", line 8753, in render
    mouse_pos = store.mas_windowutils.getMousePosRelative()
AttributeError: 'StoreModule' object has no attribute 'getMousePosRelative'
```

**Comment by Booplicate** [comment]
> AttributeError: 'StoreModule' object has no attribute 'getMousePosRelative'

The PR that adds that method hasn't been merged yet. For testing I just chucked it into a temp rpy. This shouldn't be merged 'til we get #7006 though.

**Review by ThePotatoGuy — COMMENTED** [review]
it seems to me that she blinks more often when advancing dialogue after a line completes. Also these blinks seem much faster than normal to the point where it feels like flickering. I'll hold off on a full approval until @multimokia and @jmwall24 take a look at the feel.

**Review by ThePotatoGuy — COMMENTED** [review]
would these new changes potentially cause issues if we had two monika sprites on the screen at the same time?

**Comment by Booplicate** [comment]
All Monikas would blink in sync which would look bad, but our current system doesn't allow to show 2 Monikas anyway. Are we going to do it in the future?

**Comment by ThePotatoGuy** [comment]
its an option under consideration in #5026. but we could probably just force one to be static.

**Review by ThePotatoGuy — APPROVED** [review]

**Review by multimokia — APPROVED** [review]

**Review by jmwall24 — APPROVED** [review]

**Review by Legendkiller21 — APPROVED** [review]

**Pull request merged**